modules/las/src/lib/get-las-schema.ts

```diff
@@ -1,29 +1,44 @@
-// import {Schema, Field, Float32, Uint8, FixedSizeList, MeshAttributes} from '@loaders.gl/schema';
-import {Schema, Field, MeshAttributes} from '@loaders.gl/schema';
+import {Schema, MeshAttributes, deduceMeshSchema} from '@loaders.gl/schema';
 import type {LASHeader} from './las-types';
 
 /**
  * Gets schema from PLY header
- * @param plyHeader
+ * @param lasHeader
  * @param metadata
  * @returns Schema
  */
-export function getLASSchema(plyHeader: LASHeader, attributes: MeshAttributes): Schema {
-  const fields: Field[] = [];
-
-  // if (offset.x !== undefined) {
-  //   fields.push(
-  //     new Field('POSITION', new FixedSizeList(3, new Field('xyz', new Float32())), false)
-  //   );
-  // }
-
-  // if (offset.normal_x !== undefined) {
-  //   fields.push(new Field('NORMAL', new FixedSizeList(3, new Field('xyz', new Float32())), false));
-  // }
-
-  // if (offset.rgb !== undefined) {
-  //   fields.push(new Field('COLOR_0', new FixedSizeList(3, new Field('rgb', new Uint8())), false));
-  // }
+export function getLASSchema(lasHeader: LASHeader, attributes: MeshAttributes): Schema {
+  const metadataMap = makeMetadataFromLasHeader(lasHeader);
+  const schema = deduceMeshSchema(attributes, metadataMap);
+  return schema;
+}
 
-  return new Schema(fields);
+/**
+ * Make arrow like schema metadata by LASHeader properties
+ * @param lasHeader
+ * @returns
+ */
+function makeMetadataFromLasHeader(lasHeader: LASHeader): Map<string, string> {
+  const metadataMap = new Map();
+  metadataMap.set('pointsOffset', lasHeader.pointsOffset.toString(10));
```

Comment:
Nice that you are including the metadata. Since this is las specific maybe we should use prefix `las_pointsOffset`

Reply:
Done